nixos/tests/service-runner.nix

```diff
@@ -23,7 +23,7 @@ import ./make-test-python.nix ({ pkgs, ... }: {
         machine.fail(f"curl {url}")
         machine.succeed(
             """
-            mkdir -p /run/nginx /var/spool/nginx/logs
+            mkdir -p /run/nginx /var/log/nginx /var/cache/nginx
```

Comment:
Why do we have to create the directories in the test?
I just moved system to the latest nixos-unstable bump and the log directory wasn't created automatically. I think we shouldn't fixup the tests like this. It is hiding the real errors...

Reply:
I think the comment is still true but I was able to solve my issue. It was unrelated to this change.

Reply:
Service nginx disabled by default:
```
systemd.services.nginx.enable = false;
```
Folders `/run/nginx`, `/var/log/nginx` and `/var/cache/nginx` not created automatically.

Reply:
That `service-runner` module seems to be a "startup script generator", by parsing unit attributes.

It was [added in 2013](https://github.com/nixos/nixpkgs/commit/2b0aea17934fda4aca24d4b6b99f0b9c24486ff7#diff-150236180cfb973955f1372554008dd5), and hasn't really seen a lot of changes to get on par with systemd.

I'm not sure who is using the service-runner thing today (@roberth maybe?), but I expect to not really work with many of the modules we use, and the systemd features we make use of.

Creating the directories was mostly added as a hack to not break the test, and I'd say adding support for all of systemd's features into a startup script generator was out of scope for this PR.

Reply:
So the confusion here is that this is not a test for nginx but a test for the `service-runner` script.

`service-runner` is a mostly convenience for testing, but you could use it experimentally for other purposes. I don't use it for anything critical. I wrote this test to avoid bitrot in `service-runner`. I picked nginx because it stresses the argument handling in `service-runner`.

As flokli said, this changed line is a perfectly acceptable solution.